android/src/main/java/com/lwansbrough/RCTCamera/RCTCameraModule.java

```diff
@@ -571,6 +587,70 @@ public void onShutter() {
      * concurrently which would blow the memory (esp on smaller devices), and slow things down.
      */
     private synchronized void processImage(MutableImage mutableImage, ReadableMap options, Promise promise) {
+        /*
+        DEVELOPMENT; Use this to debug cropped barcode
+        RCTCamera settings = RCTCamera.getInstance();
+        if ( settings.barcodeFinderVisible() && settings.barcodeFinderHeight() != 0 && settings.barcodeFinderWidth() != 0){
+
+            // find rotation that will be used
+            int deviceRotation = settings.getActualDeviceOrientation();
+            int rotationIndex = 0;
+            switch(deviceRotation){
+                case 0: // portrait
+                    rotationIndex = 6;
+                    break;
+                case 1: // landscape left
+                    rotationIndex = 1;
+                    break;
+                case 2: // portrait upside down
+
+                    break;
+                case 3: // landscape right
+                    rotationIndex = 3;
+                    break;
+            }
+            // rotate
```

Comment:
I believe as discussed in https://github.com/react-native-community/react-native-camera/pull/2766 it is easier to expose the method to the `react-native` developers to create the [`PlanarYUVLuminanceSource`][1] instead of handling all the different scenarios and orientations... Many users will scan barcodes only in Landscape/FrontCamera or Portrait/FrontCamera... and maybe using `com.google.zxing` libraries is not the best choice for offering all this functionalities.. so I would be happy to discuss about this, but I will focus on the android side.
Sooner or later this functionality will break and we may not have the manpower to handle the mantainance.

[1]: https://zxing.github.io/zxing/apidocs/com/google/zxing/PlanarYUVLuminanceSource.html

```C++
            switch(deviceRotation){
                case 0: // portrait
                    rotationIndex = 6;
                    break;
                case 1: // landscape left
                    rotationIndex = 1;
                    break;
                case 2: // portrait upside down

                    break;
                case 3: // landscape right
                    rotationIndex = 3;
                    break;
            }
            // rotate
            try{
                mutableImage.rotate(rotationIndex);
            } catch (MutableImage.ImageMutationFailedException e) {
                android.util.Log.e("RCTCamera","Rotate temp image",e);
                promise.reject("failed to rotate file", e);
                return;
            }
```

You can still consider working on your own library and include this functionality, as the issue here is that the owner of the code then mantains it... as other developers may not be as productive handling issues on code they did not write, having a simple api may really help a lot.